scripts/process_most_recent.py

```diff
@@ -0,0 +1,238 @@
+"""
+===================
+process_most_recent
+===================
+
+This command processes the most recent CPCIR data that is available.
+"""
+import argparse
+from datetime import datetime, timedelta
+import subprocess
+from tempfile import NamedTemporaryFile
+
+
+def run(args):
+    """
+    Process input files.
+
+    Args:
+        args: The namespace object provided by the top-level parser.
+    """
+    end_time = datetime.now()
+    start_time = end_time - timedelta(days=args.days)
+    start_time = start_time.strftime("%Y-%m-%dT%H:%M:%S")
+    end_time = end_time.strftime("%Y-%m-%dT%H:%M:%S")
+    output = args.output
+    input_type = "cpcir"
+    model = args.model
+
+    command = f"ccic process {model} {input_type} {output} {start_time} {end_time}"
+
+    input_path = args.input_path
+    if input_path is not None:
+        command += f" --input_path input_path"
+
+    targets = args.targets
+    command += f" --targets {' '.join(targets)}"
+    tile_size = args.tile_size
+    command += f" --tile_size {tile_size}"
+    overlap = args.overlap
+    command += f" --overlap {overlap}"
+    device = args.device
+    command += f" --device {device}"
+    precision = args.precision
+    command += f" --precision {precision}"
+    output_format = args.output_format
+    command += f" --output_format {output_format}"
+    credible_interval = args.credible_interval
+    command += f" --credible_interval {credible_interval}"
+    n_processes = args.n_processes
+    command += f" --n_processes {n_processes}"
+
+    roi = args.roi
+    if roi is not None:
+        command += f" --roi {' '.join(roi)}"
+
+    if args.inpainted_mask:
+        command += " --inpainted mask"
+
+    transfer = args.transfer
+    if transfer is not None:
+        command += f" --transfer {transfer}"
+
+    input_path = args.input_path
+    if input_path is not None:
+        command += f" --input_path {input_path}"
+
+    command += " --failed"
+    command += " --database_path processing_cpcir.db"
+
+    slurm = args.slurm
+    if slurm is not None:
+        with open(args.slurm, "r") as inpt:
+            script = inpt.read()
+        script = script.format(command=command)
+        with NamedTemporaryFile() as script_file:
+            script_file = "process_most_recent.sh"
+            with open(script_file, "w") as output:
+                output.write(script)
+            slurm_command = f"sbatch {script_file}"
+            subprocess.run(slurm_command.split(" "))
```

Comment:
Yes, this needs to be executed on the cluster to submit jobs. However, this works with any Python installation since it only relies on standard features. It can be executed as a one-line command appended to ``ssh``.